tests/testsuite/version.rs

```diff
@@ -53,10 +60,16 @@ fn version_works_with_bad_target_dir() {
 fn verbose() {
     // This is mainly to check that it doesn't explode.
     cargo_process("-vV")
-        .with_stdout_contains(&format!("cargo {}", cargo::version()))
-        .with_stdout_contains("host: [..]")
-        .with_stdout_contains("libgit2: [..]")
-        .with_stdout_contains("libcurl: [..]")
-        .with_stdout_contains("os: [..]")
+        .with_stdout_data(str![[r#"
+cargo [..] ([..] [..])
+release: [..]
+commit-hash: [..]
+commit-date: [..]
+host: [HOST_TARGET]
+libgit2: [..] (sys:[..] [..])
+libcurl: [..] (sys:[..] [..])
+os: [..]
```

Comment:
This one is a bit tricky. SSL information might be there if people build cargo with `openssl` Cargo feature. The test will fail under that circumstance. 

https://github.com/rust-lang/cargo/blob/d899ce72ecff0a7486ca5efa5347f719e5a44214/src/bin/cargo/cli.rs#L260-L269

The official Rust toolchain distribution uses the [`all-static`](https://github.com/rust-lang/cargo/blob/d899ce72ecff0a7486ca5efa5347f719e5a44214/Cargo.toml#L258) Cargo feature so it won't affect any CI jobs ran in rust-lang/rust repo. However, some distro packagers manage their own dependencies and don't like vendoring openssl or libgit2. They might be impacted by this when running the test suite without `all-static` feature.

One way to solve this is using make the assertion unordered by calling `.unordered()`, and put `...` at the end of the literal. And the same we may want to remove `str![]` for this.

Reply:
> One way to solve this is using make the assertion unordered by calling `.unordered()`, and put `...` at the end of the literal. And the same we may want to remove `str![]` for this.

Take `tests/testsuite/bench.rs` for example: isn't `str![]` needed by `.unordered()`?

https://github.com/rust-lang/cargo/blob/103e6758f11af08b7bd3bc6ebe7e7b270bfc7233/tests/testsuite/bench.rs#L1608-L1619

Is there any testsuite that already uses this technique that I can refer to?

Reply:
True and not ture. [`unordered()`](https://docs.rs/snapbox/latest/snapbox/data/struct.Data.html#method.unordered) is a method of `Data`. Haven't tested it but you could probably do something like:

```rust
with_stdout_data(format!("<omit>").into().unordered()))
```

Reply:
BTW, see https://github.com/rust-lang/cargo/issues/14039 "6. Contains / Does not Contains deprecations"

> A `contains` can be modeled by surrounding a entry with `...` (text) or `"...",` (json lines).  The question is whether we should still do this or use a different method.
> ....
> - Multiple `contains` can be merged into a single `eq` with `..` interspersed (or one `...` and `.unordered()`).  For example, see 3054936cab3d4627d5d1d878426c4fd88fab41f8
>   - When using `unordered`, likely string literals, rather than `str![]` should be used, as these can't correctly be updated

Reply:
addressed at commit 5d44b52

Reply:
Is `unordered` necessary? From the source code, the order of lines of version information should be idempotent:

https://github.com/rust-lang/cargo/blob/d899ce72ecff0a7486ca5efa5347f719e5a44214/src/bin/cargo/cli.rs#L208-L217